tutor/commands/config.py

```diff
@@ -48,7 +48,9 @@ def save(
     unset_vars: List[str],
     env_only: bool,
 ) -> None:
-    config = interactive_config.load_user_config(context.root, interactive=interactive)
+    config = interactive_config.load_user_config(
+        context.root, interactive=interactive, dev_context=context.is_dev()
+    )
```

Comment:
I'd rather avoid adding new methods to the context objects. Instead, I suggest to get rid of the `interactive.load_user_config` function and to add a `run_for_prod=None` argument to `interactive.ask_questions`. 

```
def ask_questions(config: Config, run_for_prod=None):
    if run_for_prod is None:
        # ask "Are you configuring a production platform?"
    ...
```

Each `quickstart` method will make a call to `ask_questions` instead of `load_user_config`:

```
def quickstart(...):
    config = tutor_config.load_minimal(root)
    if interactive:
        # run_for_prod will be:
        # None in local
        # False in dev
        # True in k8s
        interactive.ask_questions(config, run_for_prod=...)
```

Reply:
Much cleaner.

In the existing implementation, quickstart's call to `config.save` causes these lines to get executed:
```python
if not env_only:
    tutor_config.save_config_file(context.root, config)
# Reload configuration, without version checking
config = tutor_config.load_full(context.root)
env.save(context.root, config)
```
Since we're not calling `config.save` any more, should I also copy the equivalent of those lines in to `local.quickstart` and `dev.quickstart`?

Reply:
Update: I did this. This block appears in each `quickstart`:
```python
    click.echo(fmt.title("Interactive platform configuration"))
    config = tutor_config.load_minimal(context.obj.root)
    if not non_interactive:
        interactive_config.ask_questions(config, run_for_prod=...)
    tutor_config.save_config_file(context.obj.root, config)
    config = tutor_config.load_full(context.obj.root)
    tutor_env.save(context.obj.root, config)
```
Kinda feels like it should be factored out.